<launch>

	<!-- NOTE -->
	<!-- Two nodes canNOT access the same GPIO line -->
	<!-- The first node to open a line blocks other processes from accessing it -->

	<!-- START A GROUP WITH NAMESPACE "template" -->
	<group ns="ns_for_group">
		<!-- LAUNCH A "Template Encoder Read Multi Threaded" NODE -->
		<node
			pkg    = "asclinic_pkg"
			name   = "template_encoder_read_multi_threaded"
			output = "screen"
			type   = "template_encoder_read_multi_threaded"
			>
			<param
				name   = "line_number_for_motor_left_channel_a"
				value  = "133"
			/>
			<param
				name   = "line_number_for_motor_right_channel_a"
				value  = "105"
			/>
			<param
				name   = "delta_t_for_publishing_counts"
				value  = "0.1"
			/>
			<!-- NOTE: The following parameters are purely for the convenience of testing. -->
			<param
				name   = "time_in_seconds_to_drive_motors"
				value  = "10.0"
			/>
			<param
				name   = "drive_motor_target_speed"
				value  = "30"
			/>
		</node>

		<!-- LAUNCH A "Template I2C Internal" NODE -->
		<node
			pkg    = "asclinic_pkg"
			name   = "template_i2c_internal"
			output = "screen"
			type   = "template_i2c_internal"
		/>

	</group>

</launch>
